samples/subsys/usb/cdc_acm/prj.conf

```diff
@@ -3,6 +3,7 @@ CONFIG_GPIO=y
 CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
 CONFIG_USB_DEVICE_PRODUCT="Zephyr CDC ACM sample"
+CONFIG_USB_DEVICE_PID=0x001
```

Comment:
After discussion with @carlescufi:
Zephyr has its own VID so assigning PIDs is just a matter of us keeping track of them.
Proposed mechanism is to have unique PIDs for all USB samples and keep them in usb_pid.h from which docs will be generated. Like:
`#define ZEPHYR_USB_PID_CDC_ACM_SAMPLE = 0x0001`
`CONFIG_USB_DEIVICE_PID` will be skipped when setting the PID in those cases (it should be somehow made clear in menuconfig).
I will add a commit with proposal.

Reply:
This looks good keep those PIDs in `usb_pid.h`, I just do not sure we need those assigned for standard serial.

Reply:
The .inf file uses those PIDs to set names for COM ports. We could use default `0x100` for simple CDC, but unique PIDs will probably be useful in future. So I thought it's better to do it now.

Reply:
This patch is not filling up properly. I suspect Windows OS can not otherwise distinguish between samples. I do not think it is good to waste PID just because Windows OS driver for CDC ACM class is so bad.

Reply:
Yes, Windows chooses driver based on VID and PID.

What I meant is that we could use `0x100` if we only care about CDC ACM support in Windows. Using one PID causes all examples to be detected as CDC ACM and breaks HID examples. I would prefer not to do that.
Second option is to add `0x101` PID and use it with all CDC ACM examples, but since we already add a new PID I think it's better to have unique ones for each sample.
Future examples might require specific Windows drivers too.
Unique PID also allows to define different names for samples and, in case of hid-cdc, interfaces - this improves the experience.
I fail to see a problem with allocating many PIDs considering Zephyr has its own VID.

Reply:
@jfischer-phytec-iot 
>I do not think it is good to waste PID just because Windows OS driver for CDC ACM class is so bad.

The thing is, what do you mean by "waste"? We instruct our users *not* to use the Zephyr VID in our doc:

>When a vendor integrates the Zephyr USB subsystem into a product, the vendor must use the USB Vendor and Product ID assigned to them. A vendor integrating the Zephyr USB subsystem in a product must not use the Vendor ID of the Zephyr project.

Meaning that the Zephyr PIDs are reserved for us, upstream Zephyr, so I see no issue in actually using them in the samples, since those are actually the only applications that will use them? I think samples upstream should be the only users of Zephyr VID/PID.

Reply:
@carlescufi I am not against assigning dedicated PID to each sample, just the motives are not appealing to me. I also suggest that the TSC should formally approves the changes.

Reply:
OK! I will add a TSC label to this PR